src/main/scala/com/dwolla/config/package.scala

```diff
@@ -1,28 +1,33 @@
 package com.dwolla
 
-import cats._
-import cats.syntax.all._
-import com.dwolla.fs2aws.kms._
+import cats.effect.*
+import cats.syntax.all.*
+import com.amazonaws.kms.{CiphertextType, KMS}
+import fs2.compression.Compression
+import monix.newtypes.NewtypeWrapped
+import mouse.all.*
 import pureconfig.ConfigReader
-import shapeless.tag
-import shapeless.tag.@@
+import smithy4s.Blob
+import smithy4s.aws.{AwsClient, AwsEnvironment}
 
 package object config {
   private[this] val secureStringRegex = "^SECURE: (.+)".r
 
-  def SecureReader[F[_] : Monad](decryptionClient: KmsAlg[F]): ConfigReader[F[SecurableString]] =
-    ConfigReader[String].map {
-      case secureStringRegex(cryptotext) =>
-        for {
-          bytes <- decryptionClient.decrypt(cryptotext)
-        } yield tagSecurableString(bytes)
-      case s => tagSecurableString(s).pure[F]
-    }
+  def SecureReader[F[_] : Async : Compression](awsEnv: AwsEnvironment[F]): Resource[F, ConfigReader[F[SecurableString]]] =
```

Comment:
The [Smithy4s documentation recommends against](https://disneystreaming.github.io/smithy4s/docs/overview/sharing-specs#a-word-of-warning) "treating Smithy4s-generated code as publishable library-material." However, this interface doesn't expose the generated KMS classes in any way; rather it accepts an `AwsEnvironment[F]` (something published by smithy4s) and returns a `ConfigReader[F[SecurableString]]`, where `SecurableString` is defined by this library. As such I think for MiMa purposes we can treat the code in `com.amazonaws.kms` as private to this library, meaning we shouldn't need to worry about its binary compatibility. 

(It would be better to move `com.amazonaws.kms` to another package, e.g. `com.dwolla.aws.kms` or something like that, to minimize the chances of a clash with another library or generated code using `com.amazonaws.kms`. See the TODO in `build.sbt`.)